auto_gromos/gen_links.py

```diff
@@ -0,0 +1,159 @@
+import itertools
+from collections import defaultdict
+import vermouth
+from vermouth.forcefield import ForceField
+from vermouth.ffinput import read_ff
+from vermouth.molecule import Interaction
+import networkx as nx
+from auto_gromos import DATA_PATH
+from auto_gromos.assign_functional_groups import delete_atomname
+from auto_gromos.gen_bonded_interactions import gen_bonded_interactions
+from auto_gromos.ff_writers import write_links
+
+def _treat_link_atoms(molecule, link, inter_type):
+    # the uncommented statement does not work because node and
+    # atom name are couple for blocks, which is debatably useful
+    #atom_names = list(nx.get_node_attributes(block, 'atomname'))
+    atom_names = nx.get_node_attributes(molecule, "atomname")
+    resids = nx.get_node_attributes(molecule, "resid")
+    new_interactions = defaultdict(list)
+    for inter_type in link.interactions:
+        for interaction in link.interactions[inter_type]:
+            new_atoms = []
+            for atom in interaction.atoms:
+                order = resids[atom] - 1
+                prefix = "".join(["+" for _ in range(0, order)])
+                new_name = prefix + atom_names[atom]
+                new_atoms.append(new_name)
+                attrs = molecule.nodes[atom]
+                link.add_node(new_name, **attrs)
+                nx.set_node_attributes(link, {new_name: order}, "order")
```

Comment:
```suggestion
                link.add_node(new_name, order=order, **attrs)
```
Should you check whether there already is a node with new_name?